### ISSUE TITLE
Cannot infer constructor arguments with diamond inheritance

### ISSUE THREAD

**Issue by pylint-bot** [PyCQA/pylint#354]
Originally reported by: **Alexey Kotlyarov (BitBucket: [koterpillar](http://bitbucket.org/koterpillar), GitHub: @koterpillar?)**

---

Pylint does not infer the correct constructor of a derived class when using diamond inheritance.

Sample code:

``` python
class A(object):
    def __init__(self):
        pass


class B(A):
    def __init__(self, arg):
        pass


class C(A):
    pass


class D(C, B):
    pass


D(arg=1)
```

Pylint output:

```
E: 19, 0: Unexpected keyword argument 'arg' in constructor call (unexpected-keyword-arg)
```

Versions:
- Pylint 1.3.1
- Python 3.4.1
- Python 2.7.8
  (both Python 2 and 3 are affected)

---
- Bitbucket: https://bitbucket.org/logilab/pylint/issue/354
